roles/default/kiali-deploy/templates/kubernetes/service.yaml

```diff
@@ -1,14 +1,9 @@
 apiVersion: v1
 kind: Service
 metadata:
-  name: kiali
+  name: {{ kiali_vars.deployment.instance_name }}
```

Comment:
There are some edge cases with naming kube Services. Specifically they need to conform to the [DNS label standard](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-label-names). If the value of `instance_name` is being used for the `Service` name, either `instance_name` should also conform to the label standard or be messaged for the `Service` name.

Reply:
Right - the assumption is the name will follow conventions. We'll document this (I actually meant to, but forgot)... but we won't do any massaging - its simple enough to require people to use names that follow the standards for both DNS and k8s in general.

Reply:
I'm going to update the kiali.io doc PR for this. I'll also see if I can put in a simple "test" in the operator task to check for any problems in the instance_name field... It should be trivial to add a regex check in the ansible task to check that the instance_name doesn't violate any obvious naming rules.

Reply:
added blurb here: https://github.com/kiali/kiali.io/pull/374
I'll add the regex check shortly - it'll be a new commit to this PR

Reply:
Here's what the regex will look like - and here's how I tested the regex using ansible (this is what the test will look like in the operator) - note that I restrict the name to 50 chars (not 63) because it gives us some leeway as this instance_name is only a prefix and we need more chars to prepend to it for some resource names like our cabundle configmap and other resources.

```
  - debug:
      msg: "Bad dns name: {{ item }}"
    vars:
      str: "{{ item }}"
    when:
    - str is not regex('^(?![0-9]+$)(?!-)[a-z0-9-]{,50}(?<!-)$')
    loop:
    # good names
    - "a"
    - "ab"
    - "abc"
    - "a-c"
    - "1ab"
    - "a1b"
    - "ab1"
    - "a1234567890abcdefghujklmnopqrstuvwxyz01234567890ab"
    # bad names
    - "-"
    - "-xy"
    - "xy-"
    - "x.y"
    - "XY"
    - "a1234567890abcdefghujklmnopqrstuvwxyz01234567890abx"
```

Reply:
see: https://github.com/kiali/kiali-operator/pull/313/commits/e42dd9ada991b480d318adac5679b74db0cc48a6